xsub.go

```diff
@@ -7,18 +7,24 @@ package zmq4
 import (
 	"context"
 	"net"
+	"strings"
+	"sync"
 )
 
 // NewXSub returns a new XSUB ZeroMQ socket.
 // The returned socket value is initially unbound.
 func NewXSub(ctx context.Context, opts ...Option) Socket {
-	xsub := &xsubSocket{newSocket(ctx, XSub, opts...)}
+	xsub := &xsubSocket{newSocket(ctx, XSub, opts...), sync.RWMutex{}, make(map[string]struct{})}
```

Comment:
```suggestion
	xsub := &xsubSocket{sck: newSocket(ctx, XSub, opts...)}
```

or:
```suggestion
	xsub := &xsubSocket{
		sck:    newSocket(ctx, XSub, opts...),
		topics: make(map[string]struct{}),
	}
```
(and drop the extra `make(map[...])` of line `19`.)

Reply:
Committed in branch issue-108.